components/script/dom/filelist.rs

```diff
@@ -18,6 +18,7 @@ pub struct FileList {
 }
 
 impl FileList {
+    #[allow(unrooted_must_root)]
     fn new_inherited(files: Vec<JS<File>>) -> FileList {
```

Comment:
No, only `box unrooted` is safe in new, everything else isn't.

In this case it's the pat from the argument list that's being caught. It wasn't caught in the past since `check_fn` is not run on new_*.

This can be fixed by adding another state to the lint, "in_argument_list", which is set unconditionally by `check_fn` and turned off by `check_block`, and bailing in `check_pat` when `in_new_function && in_argument_list`